src/main/java/seedu/address/logic/commands/DoneCommand.java

```diff
@@ -0,0 +1,41 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.commons.core.Messages.MESSAGE_NOT_IN_SERVE_MODE;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+
+/**
+ * Exits the serve mode.
+ */
+public class DoneCommand extends Command {
+    public static final String COMMAND_WORD = "done";
+
+    public static final String MESSAGE_SUCCESS = "Exited from Serve Mode. ";
+
+    /**
+     * Executes the command and returns the result message.
+     *
+     * @param model {@code Model} which the command should operate on.
+     * @return feedback message of the operation result for display
+     * @throws CommandException If an error occurs during command execution.
+     */
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+
+        if (!model.isServeMode()) {
+            throw new CommandException(MESSAGE_NOT_IN_SERVE_MODE);
+        }
+
+        model.exitsServeMode();
+        return new CommandResult(MESSAGE_SUCCESS);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this || other instanceof DoneCommand) {
```

Comment:
I think you may just `return other == this || other instanceof DoneCommand` to make this less covoluted

Reply:
Oh right. Thanks for the suggestion!